sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/PushNotifier.java

```diff
@@ -145,23 +160,40 @@ private static Notification getOldNotification(Notification.Builder builder) {
     /**
      * Sets the color in the notification builder if the property is set in the intent.
      *
+     * @param context    The current context.
      * @param pushIntent The push intent.
      * @param builder    The builder to modify.
      */
-    private static void setColor(Intent pushIntent, Notification.Builder builder) {
+    private static void setColor(Context context, Intent pushIntent, Notification.Builder builder) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return;
+        }
+
+        /* Check custom color from intent. */
         String colorString = PushIntentUtils.getColor(pushIntent);
         if (colorString != null) {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            try {
                 builder.setColor(Color.parseColor(colorString));
+                return;
+            } catch (IllegalArgumentException e) {
+                AppCenterLog.error(LOG_TAG, "Invalid color sting.", e);
```

Comment:
typo.
Should mention  " received in push payload".
Also is it really throwing exception right now?

Reply:
https://github.com/aosp-mirror/platform_frameworks_base/blob/ed631bf5aad3f61b859d37446cd5d152924bd92d/graphics/java/android/graphics/Color.java#L1400